server/controllers/configurationsController/configurationsController.test.js

```
@@ -0,0 +1,166 @@
+import { MongoError } from 'mongodb'
+import ConfigurationsController from '.'
+import { createRequestWithUser, createResponse } from '../../../test/fixtures'
+
+describe('ConfigurationsController', () => {
+  describe(ConfigurationsController.create, () => {
+    const newConfiguration = { body: { name: 'new matrix', owner: 'owl' } }
+    const request = createRequestWithUser(newConfiguration)
+    const response = createResponse()
+
+    it('sends a status of 200 with a data property when successful', async () => {
+      request.app.locals.appDb.insertOne.mockResolvedValueOnce({
+        insertedId: 'eagle',
+      })
+
+      await ConfigurationsController.create(request, response)
+
+      expect(response.status).toHaveBeenCalledWith(200)
+      expect(response.json).toHaveBeenCalledWith({
+        data: { insertedId: 'eagle' },
+      })
+    })
+
+    it('returns a 400 error with an error message when unsuccessful', async () => {
+      request.app.locals.appDb.insertOne.mockImplementation(() => {
+        new Error()
+      })
```

I'm not sure this is doing what we expect. This error is not being returned or thrown by the mocked function. Are you trying to return the error, throw the error, or do you want the promise to be rejected? If you're looking for the promise to be rejected you can use `mockRejectedValueOnce`.

---

[f11e811](https://github.com/AMP-SCZ/dpdash/pull/305/commits/f11e811a37bb7df8cc158a8d0e061615e8ec535d)